src/extension/background-script/actions/payments/listByAccount.ts

```diff
@@ -1,24 +1,71 @@
-import type { MessagePaymentListByAccount } from "~/types";
+import { ConnectorTransaction } from "~/extension/background-script/connectors/connector.interface";
+import state from "~/extension/background-script/state";
+import type {
+  DbPayment,
+  MessagePaymentListByAccount,
+  Transaction,
+} from "~/types";
 
 import db from "../../db";
 
+// @Todo: renmae, due to fact that "byAccount" will be default
+
 const listByAccount = async (message: MessagePaymentListByAccount) => {
-  const { accountId } = message.args;
+  const accountId = state.getState().currentAccountId;
+
   // TODO: Add pagination instead of limiting to 2121
   const limit = message?.args?.limit || 2121;
 
-  const payments = await db.payments
+  const connector = await state.getState().getConnector();
+
+  // get IndexDB payments
+  const dbPayments = await db.payments
     .toCollection()
     .filter((p) => p.accountId === accountId)
     .limit(limit)
     .reverse()
     .sortBy("createdAt");
 
-  return {
-    data: {
-      payments,
-    },
-  };
+  try {
+    const result = await connector.getTransactions();
+
+    // map IndexDB payment data onto ConnectorTransaction
+    const transactions: Transaction[] = result.data.transactions.map(
```

Comment:
Because there are currently limits on the number of invoices (and TBC: maybe we do not do the getTransactions() implementation for all connectors) maybe we should create a list from both the DB and the result of getTransactions, only merging the transactions that exist in both. What do you think?

Reply:
@reneaaron mentioned this in the last community call. I think this gets to confusing for the user.  E.g.: the first 100 (which is the result set for many unpaginated connector result sets) comes from the connector, the rest from IndexDB. Users might wonder why older transactions are missing? Also explaining this in the UI is not really cool.

Talking about the pagination topic René mentioned he is quite surprised the missing pagination for incoming transactions was never really an issue for users so far. It seems only latest are relevant.

I would tend to implement as is, implement pagination for connectors that support it (our lndhub.go doesn't but it's in our control), LND does support it, and implement the rest as it will be supported in the future. 

@reneaaron @rolznz what do you think?

Reply:
Currently we see no transactions at all that happened outside of the extension, so I think seeing the 100 latest transactions is a huge improvement.

I am just worried if we try implement all the connectors it will delay this getting merged. I think we should focus on the most important connectors and improve in small steps, especially when we have to rely on other dependencies to also be updated.

Reply:
@reneaaron @rolznz what do you propose? Have two concurrent logics, getting transactions from the connector for lndhub (as a first implementation) and all others with the current logic of getting data from IndexDB? I think this restricts us in refactoring the code as well. @im-adithya offered to help out implementing the connectors. What do you guys think?

Reply:
I see your point, but I don't feel good about having separate logics for connectors and as well would like to have the freedom in refactoring the interfaces/data in the transactions list to make it clean.